app/services/discovery_engine/sync/delete.rb

```diff
@@ -1,46 +1,21 @@
 module DiscoveryEngine::Sync
   class Delete < Operation
     def call
-      lock.acquire
-
-      unless version_cache.sync_required?
+      super do
```

Comment:
It feels a bit unconventional to call super with a block like this. I wonder if you might have code that is a little easier to read if the parent class defined a method of sync rather than call that you called?

You could also pass that an argument of type rather than having that as an instance method as I found the purpose of that a little cryptic.